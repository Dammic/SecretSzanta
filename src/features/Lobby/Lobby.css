@import 'variables/variables.css';

.gameListContainer {
    display: flex;
    height: calc(100% - var(--header-height));
}

.firstColumn {
    /* 35px is margin top and 35px is margin bottom */
    height: calc(100% - 35px - 35px);
    background-color: var(--game-list-first-column-bg);
    overflow: auto;
    width: 70%;
    margin: 35px 0 35px 10px;
    padding: 10px 10px 0 10px;
}

.secondColumn {
    margin-left: 10px;
    margin-right: 10px;
    width: 40%;
    height: 100%;
    display: flex;
    flex-direction: column;
    @media (--tablet) {
        width: 30%;
    }
}

.lobbyChat {
    position: relative;
    margin-top: 15px;
    margin-left: 2px;
    height: 220px;
    width: auto;
    flex-shrink: 0;
}
