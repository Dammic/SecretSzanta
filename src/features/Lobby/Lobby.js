import React from 'react'
import { GameList } from './components/GameList'
import { TopNavbar } from './components/TopNavbar'
import { PlayersList } from './components/PlayersList'
import Chat from '../Chat/Chat'

import styles from './Lobby.css'

export const Lobby = () => (
    <div className={styles.gameListContainer}>
        <div className={styles.firstColumn}>
            <GameList />
        </div>
        <div className={styles.secondColumn}>
            <TopNavbar />
            <PlayersList />
            <Chat className={styles.lobbyChat} />
        </div>
    </div>
)
