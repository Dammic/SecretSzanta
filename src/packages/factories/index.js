export { arrayOf } from './arrayOf'
export { aMessage } from './messages'
export { aRoom, aRoomsList, aPlayer, aPlayersList } from './lobby'
