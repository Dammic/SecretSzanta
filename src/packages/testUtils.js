import renderer from 'react-test-renderer';
import ShallowRenderer from 'react-test-renderer/shallow';

const shallowRenderer = new ShallowRenderer()

export const expectMatchingSnapshot = (element) => {
    const renderedElement = renderer.create(element)
    expect(renderedElement).toMatchSnapshot()
    renderedElement.unmount()
}

export const expectShallowMatchingSnapshot = (element) => {
    const renderedElement = shallowRenderer.render(element)
    expect(renderedElement).toMatchSnapshot()
    renderedElement.unmount()
}
