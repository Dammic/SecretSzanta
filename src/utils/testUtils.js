import { mount } from 'enzyme'

export const expectMatchingSnapshot = (element) => {
    const renderedComponent = mount(element)
    expect(renderedComponent).toMatchSnapshot()
}
