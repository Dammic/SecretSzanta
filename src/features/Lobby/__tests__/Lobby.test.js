import React from 'react'
import { shallow } from 'enzyme'
import { expectShallowMatchingSnapshot } from 'packages/testUtils'
import { Lobby } from '../Lobby'

describe('<Lobby />', () => {
    it('matches snapshot', () => {
        expectShallowMatchingSnapshot(<Lobby />)
    })
})
