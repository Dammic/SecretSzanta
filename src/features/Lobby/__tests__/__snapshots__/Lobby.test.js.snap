// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Lobby /> matches snapshot 1`] = `
<div
  className="gameListContainer"
>
  <div
    className="firstColumn"
  >
    <Connect(GameList) />
  </div>
  <div
    className="secondColumn"
  >
    <Connect(TopNavbar) />
    <Connect(PlayersList) />
    <Chat
      className="lobbyChat"
    />
  </div>
</div>
`;
